sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -622,6 +622,31 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
     }
   }
 
+  test("save tsv with tsv suffix") {
+    withTempDir { dir =>
+      val csvDir = new File(dir, "csv").getCanonicalPath
+      val cars = spark.read
+        .format("csv")
+        .option("header", "true")
+        .load(testFile(carsFile))
+
+      cars.coalesce(1).write
+        .option("header", "true")
+        .option("fileExtension", ".tsv")
+        .option("delimiter", "\t")
```

Comment:
When `delimiter` is set to `\t`, is it still a CSV? : )

Ref: https://en.wikipedia.org/wiki/Tab-separated_values